Automattic/jetpack
# Forms: add field id setting

This PR adds a new text control to the field block's inspector, allowing to change the name/ID for the input.

## Proposed changes:
Add the name/ID setting on the inspector's Advance section for form input blocks
<img width="286" alt="image" src="https://user-images.githubusercontent.com/157240/235969786-2371de51-e8d7-4dd1-b7a1-36aebaedffd6.png">

This is replacing our per-input implementation where it would be referred as "Unique CSS ID".

### Other information:


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Insert a form on a post. Select any input and check the inspector's Advanced section, you should be able to change the input's name/ID.
It should be restricted to alphanumeric, dash and underscore (this restriction can be bypassed using the code editor).
Look at the published post, see that the input has the name/ID set for it.
